Buildship setup fails with RepositoryNotFoundException

<!--- 
Please follow the instructions below. 

If you have a general question, please ask it on the forum: https://discuss.gradle.org/c/help-discuss/buildship
-->

<!--- Provide a brief summary of the issue in the title above -->

### Expected Behavior
Following the [Setup Instructions](https://github.com/eclipse/buildship/blob/master/docs/development/Setup.md), I should be able to install Eclipse in the Commiters Edition with the buildship project enabled.

### Current Behavior
After hitting install, the IDE launches but the buildship project is not opened and no setup tasks are executed. When going to **Help -> Perform Setup Tasks** a window with an internal error `org.eclipse.jgit.errors.RepositoryNotFoundException` pops up.

``` bash
Java.lang.Exception: org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: /home/fiete/Documents/development/buildship/git
	at org.eclipse.oomph.setup.git.impl.GitCloneTaskImpl.isNeeded(GitCloneTaskImpl.java:861)
	at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.initNeededSetupTasks(SetupTaskPerformer.java:2285)
	at org.eclipse.oomph.setup.ui.wizards.ConfirmationPage.initNeededSetupTasks(ConfirmationPage.java:355)
	at org.eclipse.oomph.setup.ui.wizards.ConfirmationPage.enterPage(ConfirmationPage.java:259)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizard.pageChanged(SetupWizard.java:658)
	at org.eclipse.jface.wizard.WizardDialog$5.run(WizardDialog.java:1518)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.wizard.WizardDialog.firePageChanged(WizardDialog.java:1515)
	at org.eclipse.jface.wizard.WizardDialog.update(WizardDialog.java:1325)
	at org.eclipse.jface.wizard.WizardDialog.updateForPage(WizardDialog.java:1257)
	at org.eclipse.jface.wizard.WizardDialog.lambda$3(WizardDialog.java:1222)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:74)
	at org.eclipse.jface.wizard.WizardDialog.showPage(WizardDialog.java:1222)
	at org.eclipse.jface.wizard.WizardDialog.nextPressed(WizardDialog.java:917)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.oomph.util.ReflectUtil.invokeMethod(ReflectUtil.java:119)
	at org.eclipse.oomph.util.ReflectUtil.invokeMethod(ReflectUtil.java:149)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizardPage.gotoPage(SetupWizardPage.java:225)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizardPage.gotoNextPage(SetupWizardPage.java:207)
	at org.eclipse.oomph.setup.ui.wizards.VariablePage$6.run(VariablePage.java:699)
	at org.eclipse.oomph.ui.UIUtil$4.run(UIUtil.java:534)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5101)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4584)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizard.openDialog(SetupWizard.java:776)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizard$Updater.openDialog(SetupWizard.java:1802)
	at org.eclipse.oomph.setup.ui.wizards.SetupWizard$Updater.perform(SetupWizard.java:1818)
	at org.eclipse.oomph.setup.presentation.handlers.PerformHandler.run(PerformHandler.java:52)
	at org.eclipse.oomph.setup.presentation.handlers.AbstractDropdownItemHandler.execute(AbstractDropdownItemHandler.java:51)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:317)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:251)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5895)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1520)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:5126)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4581)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
Caused by: org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: /home/fiete/Documents/development/buildship/git
	at org.eclipse.jgit.lib.BaseRepositoryBuilder.build(BaseRepositoryBuilder.java:627)
	at org.eclipse.jgit.api.Git.open(Git.java:93)
	at org.eclipse.jgit.api.Git.open(Git.java:73)
	at org.eclipse.oomph.setup.git.impl.GitCloneTaskImpl.isNeeded(GitCloneTaskImpl.java:819)
	... 79 more
```

This makes sense, since the repository was not cloned into that location:
``` bash
fiete@ubu:~/Documents/development/buildship$ tree -L 2
.
├── git
│   └── tooling-local.target
├── ide
│   └── eclipse
└── ws

4 directories, 1 file
```

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
I would like to be able to contribute to buildship. For that I need to install the Eclipse IDE to run the project and it's tests.

### Steps to Reproduce
I am following the [Setup Instructions](https://github.com/eclipse/buildship/blob/master/docs/development/Setup.md).
![1-eclipse-commiters](https://user-images.githubusercontent.com/41323592/163798369-23c2d484-f950-48b0-8747-0cb8c30c6e39.png)
![2-eclipse-projects](https://user-images.githubusercontent.com/41323592/163798379-96334313-c3b8-4dec-a486-0d846d67f424.png)
![3-variables](https://user-images.githubusercontent.com/41323592/163798388-de9ceb07-cb93-44ef-9a47-8c8715efe824.png)
![4-confirmation](https://user-images.githubusercontent.com/41323592/163798394-a896c7da-f495-456e-bc7a-138c327a2731.png)

Note that I have sped up the video, however I waited for enough time for any eventual automatic setup task to complete before opening that setup menu.

https://user-images.githubusercontent.com/41323592/163798620-af90cc66-6265-4257-920e-1146eb43e95b.mp4



### Your Environment
Ubuntu 20.04, openjdks 8 and 11 from the official repositories.

--- a/docs/development/OomphImport.md
+++ b/docs/development/OomphImport.md
@@ -1,0 +1,19 @@
+# Testing buildship.setup
+By default, Oomph will use the master version at: https://raw.githubusercontent.com/eclipse/buildship/master/buildship.setup.
+
+The installer can be instructed to use a different version (e.g. via eclipse-inst.ini):
+
+```ini
+-Doomph.redirection.buildship=https://raw.githubusercontent.com/eclipse/buildship/master/buildship.setup->file:/<path to buildship checkout>/buildship.setup 
+```
+
+Windows has no eclipse-inst.ini, do this instead:
+
+```
+.\eclipse-inst-jre-win64.exe -vmargs '-Doomph.redirection.buildship=https://raw.githubusercontent.com/eclipse/buildship/master/buildship.setup->file:/<path to buildship checkout>/buildship.setup'
+```
+
+Replace `<path to buildship checkout>` with the location of your Buildship checkout.
+On the variables page, enter the URL of your fork, `https://github.com/<user>/buildship`, and the branch you want to checkout. This step can be skipped, if you only want to test changes to your `buildship.setup` (the local version will be used). 
+To redirect Eclipse to use the local `buildship.setup`, use `-Dbuildship.oomphtest.enabled=true`. It will add another variable `buildship.oomphtest.setupLocation`. Set it to `<path to buildship checkout>/buildship.setup`. 
+Without this configuration, the new Eclipse installation, will perform the setup tasks using the `buildship.setup` from the update site.

--- a/docs/development/README.md
+++ b/docs/development/README.md
@@ -4,3 +4,4 @@ The following developer documentation of Buildship is currently available:
 
 1. [Development setup](Setup.md)
 2. [Public API examples](ApiExamples.md)
+3. [Oomph Import](OomphImport.md)

--- a/samples/oomph-setup/README.md
+++ b/samples/oomph-setup/README.md
@@ -8,19 +8,12 @@ Oomph will import an example gradle project using Buildship.
 
 # Testing against the local checkout
 By default the setup tasks uses the latest released Buildship version from the update site. The eclipse
-installer can be instructed to use the local site and model by the eclipse installer with the following parameters (e.g. via eclipse-inst.ini):
+installer can be instructed to use the local site and model by building it using `gradlew :org.eclipse.buildship.site:build` and
+supplying the eclipse installer with the following parameters (e.g. via eclipse-inst.ini):
 
 ```ini
--Doomph.redirection.buildship=https://raw.githubusercontent.com/eclipse/buildship/master/buildship.setup->file:/<path to buildship checkout>/buildship.setup 
+-Doomph.redirection.bs=https://raw.githubusercontent.com/eclipse/buildship/master/org.eclipse.buildship.oomph/model/GradleImport-1.0.ecore->file:/<path to builship checkout>/org.eclipse.buildship.oomph/model/GradleImport-1.0.ecore
+-Doomph.redirection.p2.buildship=https://download.eclipse.org/buildship/updates/latest->file:/<path to builship checkout>/org.eclipse.buildship.site/build/repository
 ```
 
-Windows has no eclipse-inst.ini, do this instead:
-
-```
-.\eclipse-inst-jre-win64.exe -vmargs '-Doomph.redirection.buildship=https://raw.githubusercontent.com/eclipse/buildship/master/buildship.setup->file:/<path to buildship checkout>/buildship.setup'
-```
-
-Replace `<path to buildship checkout>` with the location of your buildship checkout.
-On the variables page, enter the URL of your fork, `https://github.com/<user>/buildship`, and the branch you want to checkout. This step can be skipped, if you only want to test changes to your `buildship.setup` (the local version will be used). 
-To redirect Eclipse to use the local `buildship.setup`, use `-Dbuildship.oomphtest.enabled=true`. It will add another variable `buildship.oomphtest.setupLocation`. Set it to `<path to buildship checkout>/buildship.setup`. 
-Without this configuration, the new Eclipse installation, will perform the setup tasks using the `buildship.setup` from the update site.
+Replace `<path to builship checkout>` with the location of your buildship checkout.

